pkg/storage/redis.go

```diff
@@ -0,0 +1,366 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	goredislib "github.com/go-redis/redis/v8"
+	"github.com/go-redsync/redsync/v4"
+	"github.com/go-redsync/redsync/v4/redis/goredis/v8"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	PONG               = "PONG"
+	RedisScanBatchSize = 1000
+	RedisMutex         = "redis-mutex"
+)
+
+func init() {
+	err := RegisterStorage(&RedisDB{})
+	if err != nil {
+		panic(err)
+	}
+}
+
+type RedisDB struct {
+	db    *goredislib.Client
+	ctx   context.Context
```

Comment:
Having as an instance variable could potentially be dangerous. E.g. if there are changes to it. I recommend removing. I recommend removing for now, and creating a new context on every request.

We'll fix this later on to do it the right way.